### ISSUE TITLE
Rust 1.38.0 can't build rust 1.38.0 out-of-the-box

### ISSUE THREAD

**Issue by he32** [rust-lang/rust#65722]
Hi,

it seems the new compiler warns about a construct which isn't warned about
with rust 1.37.0:

```
   Compiling bootstrap v0.0.0 (/usr/pkgsrc/lang/rust/work/rustc-1.38.0-src/src/bootstrap)
error: unnecessary `unsafe` block
   --> src/bootstrap/builder.rs:148:19
    |
148 |             name: unsafe { ::std::intrinsics::type_name::<S>() },
    |                   ^^^^^^ unnecessary `unsafe` block
    |
note: lint level defined here
   --> src/bootstrap/lib.rs:107:9
    |
107 | #![deny(warnings, rust_2018_idioms, unused_lifetimes)]
    |         ^^^^^^^^
    = note: `#[deny(unused_unsafe)]` implied by `#[deny(warnings)]`

error: aborting due to previous error

error: Could not compile `bootstrap`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

I'll next try the obvious fix of removing `warnings` from what's denied, but, really, this ought to have worked "out of the box".
